functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,359 @@
+package __performance
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/cluster"
+	testlog "github.com/openshift-kni/performance-addon-operators/functests/utils/log"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+)
+
+var _ = Describe("[ref_id: 40307][pao] updating performance profile", func() {
+	var workerRTNodes []corev1.Node
+	var err error
+	var net bool = true
+	var profile *performancev2.PerformanceProfile
+	testutils.BeforeAll(func() {
+		isSNO, err := cluster.IsSingleNode()
+		Expect(err).ToNot(HaveOccurred())
+		RunningOnSingleNode = isSNO
+	})
+	BeforeEach(func() {
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+		nodeLabel := testutils.NodeSelectorLabels
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("Updating performance profile for netqueues", func() {
+		It("[test_id:40308][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Network device queues Should be set to the profile's reserved CPUs count ", func() {
+			devices := make(map[string]int)
+			count := 0
+			if profile.Spec.Net != nil {
+				if profile.Spec.Net.UserLevelNetworking != nil && *profile.Spec.Net.UserLevelNetworking && len(profile.Spec.Net.Devices) == 0 {
+					By("To all non virtual network devices when no devices are specified under profile.Spec.Net.Devices")
+					err := checkDeviceSupport(workerRTNodes, devices)
+					Expect(err).ToNot(HaveOccurred())
+					for _, size := range devices {
+						if size == getReservedCPUSize(profile.Spec.CPU) {
```

Comment:
why do we need this counter and check ?  I would fail the test in the first mismatch we get since the devices we get are suppose to be supported devices

Reply:
@yanirq  The need for this occurred because in the GCP Lane the nic ens3 showed that it has support for queueresize (i came to this conclusion based on Combined queue size which it shows as 4 )  but it doesn't get resized , i could not debug it more. But the same code works on Downstream BM.